src/graphics/program-lib/programs/standard.js

```diff
@@ -1454,18 +1483,53 @@ var standard = {
                 code += "   addReflection();\n";
             }
 
+            if (options.dirLightMap) {
+                code += "   addDirLightMap();\n";
+            }
+
+            if (hasAreaLights){
+                // specular has to be accumulated differently if we want area lights to look correct
+                code += "   ccReflection.rgb *= ccSpecularity;\n";
+                code += "   dReflection.rgb *= dSpecularity;\n";
+                code += "   dSpecularLight *= dSpecularity;\n";
+
+                code += "   float roughness = max((1.0 - dGlossiness) * (1.0 - dGlossiness), 0.001);\n";
+            }
+
+            // light source shape support
+            var shapeStringMap = [];
+            shapeStringMap[LIGHTSHAPE_RECT] = 'Rect';
+            shapeStringMap[LIGHTSHAPE_DISK] = 'Disk';
+            shapeStringMap[LIGHTSHAPE_SPHERE] = 'Sphere';
```

Comment:
this should be moved outside to be initialized one time only

Reply:
moved it into a private function - making it more consistent with the standard material generator